packages/eslint-plugin/src/rules/no-misused-spread.ts

```diff
@@ -0,0 +1,185 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import * as tsutils from 'ts-api-utils';
+import * as ts from 'typescript';
+
+import {
+  createRule,
+  getConstrainedTypeAtLocation,
+  getParserServices,
+  isBuiltinSymbolLike,
+  isTypeFlagSet,
+} from '../util';
+
+type MessageIds =
+  | 'noStringSpreadInArray'
+  | 'noSpreadInObject'
+  | 'noFunctionSpreadInObject';
+
+export default createRule<[], MessageIds>({
+  name: 'no-misused-spread',
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'Disallow using the spread operator when it might cause unexpected behavior',
+      recommended: 'strict',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      noStringSpreadInArray:
+        "Using the spread operator on a string is not allowed in an array. Use `String.split('')` instead.",
```

Comment:
[Docs] Something like this? _"is not allowed"_ is a bit too prescriptive.

```suggestion
        "Using the spread operator on a string can cause unexpected behavior. Prefer `String.split('')` instead.",
```

...and similar changes for the others?

Reply:
yup, I like it